src/sagemaker/cli/compatibility/v2/modifiers/py_version.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""A class to ensure that ``py_version`` is defined when constructing framework classes."""
+from __future__ import absolute_import
+
+import ast
+
+from sagemaker.cli.compatibility.v2.modifiers.modifier import Modifier
+
+PY_VERSION_ARG = "py_version"
+PY_VERSION_DEFAULT = "py3"
+
+FRAMEWORK_MODEL_REQUIRES_PY_VERSION = {
+    "Chainer": True,
+    "MXNet": True,
+    "PyTorch": True,
+    "SKLearn": False,
+    "TensorFlow": False,
+}
+
+FRAMEWORK_CLASSES = list(FRAMEWORK_MODEL_REQUIRES_PY_VERSION.keys())
+
+MODEL_CLASSES = [
+    "{}Model".format(fw) for fw, required in FRAMEWORK_MODEL_REQUIRES_PY_VERSION.items() if required
+]
+FRAMEWORK_MODULES = [fw.lower() for fw in FRAMEWORK_CLASSES]
+FRAMEWORK_SUBMODULES = ("model", "estimator")
```

Comment:
(optional) you could define this class in the `framework_versions` module so that these constants are in one place. (in my head, Python version and framework version are sort of part of the same thing anyway)

Reply:
will discard this and combine. they can share a lot, imo